packages/kit/src/runtime/server/page/index.js

```diff
@@ -18,25 +30,317 @@ export async function render_page(event, route, options, state, resolve_opts) {
+	if (accept === 'application/json') {
+		const node = await options.manifest._.nodes[route.page]();
+		if (node.server) {
+			return handle_json_request(event, options, node.server);
 		}
 	}
 
 	const $session = await options.hooks.getSession(event);
 
-	return respond({
-		event,
-		options,
-		state,
-		$session,
-		resolve_opts,
-		route
+	const { fetcher, fetched } = create_fetch({ event, options, state, route });
+
+	try {
+		const nodes = await Promise.all([
+			// we use == here rather than === because [undefined] serializes as "[null]"
+			...route.layouts.map((n) => (n == undefined ? n : options.manifest._.nodes[n]())),
+			options.manifest._.nodes[route.page]()
+		]);
+
+		// TODO for non-GET requests, first call handler in +page.server.js
+		// (this also determines status code)
+		const leaf_node = /** @type {import('types').SSRNode} */ (nodes.at(-1));
+
+		if (!resolve_opts.ssr) {
+			return await render_response({
+				branch: [],
+				fetched,
+				page_config: {
+					hydrate: true,
+					router: true
+				},
+				status: 200,
+				error: null,
+				event,
+				options,
+				state,
+				$session,
+				resolve_opts
+			});
+		}
+
+		if (state.prerendering) {
+			// if the page isn't marked as prerenderable (or is explicitly
+			// marked NOT prerenderable, if `prerender.default` is `true`),
+			// then bail out at this point
+			const should_prerender = leaf_node.module.prerender ?? options.prerender.default;
+			if (!should_prerender) {
+				return new Response(undefined, {
+					status: 204
+				});
+			}
+
+			const mod = leaf_node.server;
+			if (mod && (mod.POST || mod.PUT || mod.DELETE || mod.PATCH)) {
+				throw new Error('Cannot prerender pages that have endpoints with mutative methods');
+			}
+		}
+
+		/** @type {Array<Loaded | null>} */
+		let branch = [];
+
+		/** @type {number} */
+		let status = 200;
+
+		/** @type {Error | null} */
+		let error = null;
+
+		/** @type {Array<Promise<import('types').JSONObject | null>>} */
+		const server_promises = nodes.map((node, i) => {
+			if (error) throw error; // if an error happens immediately, don't bother with the rest of the nodes
+			return Promise.resolve().then(async () => {
+				try {
+					const server_data = node?.server?.GET?.call(null, {
+						...event,
+						parent: async () => {
+							const data = {};
+							for (let j = 0; j < i - 1; j += 1) {
+								Object.assign(data, await server_promises[j]);
+							}
+							return data;
+						}
+					});
+
+					return server_data ? unwrap_promises(server_data) : null;
+				} catch (e) {
+					error = /** @type {Error} */ (e);
+					throw error;
+				}
+			});
+		});
+
+		/** @type {Array<Promise<Record<string, any> | null>>} */
+		const load_promises = nodes.map((node, i) => {
+			if (error) throw error;
+			return Promise.resolve().then(async () => {
+				try {
+					const server_data = await server_promises[i];
+
+					if (node?.module?.load) {
+						const data = await node?.module?.load?.call(null, {
```

Comment:
I think this can use the newly added `load_data` function (that function needs to be adjusted then though)

Reply:
`load_data` is a vestige of the existing [`load_node`](https://github.com/sveltejs/kit/blob/bde443dad85868defb9bdfdc8f0a2347a2451582/packages/kit/src/runtime/server/page/load_node.js#L27), and should probably be got rid of as it doesn't know how to inherit data from `parent`

Reply:
(though that does involve some slightly un-DRY code i suppose, for the `respond_with_error` function)